src/test/java/de/tum/cit/aet/artemis/athena/service/connectors/AthenaFeedbackSendingServiceTest.java

```diff
@@ -79,50 +97,64 @@ class AthenaFeedbackSendingServiceTest extends AbstractAthenaTest {
 
     private Feedback programmingFeedback;
 
+    private ListAppender<ILoggingEvent> asyncExceptionLogAppender;
+
+    private Logger asyncExceptionLog;
+
     @BeforeEach
     void setUp() {
-        athenaFeedbackSendingService = new AthenaFeedbackSendingService(athenaRequestMockProvider.getRestTemplate(), athenaModuleService,
-                new AthenaDTOConverterService(textBlockRepository, textExerciseRepository, programmingExerciseRepository, gradingCriterionRepository));
-
-        athenaRequestMockProvider.enableMockingOfRequests();
+        userUtilService.addUsers(TEST_PREFIX, 2, 0, 0, 0);
 
         textExercise = textExerciseUtilService.createSampleTextExercise(null);
         textExercise.setFeedbackSuggestionModule(ATHENA_MODULE_TEXT_TEST);
-        when(textExerciseRepository.findWithGradingCriteriaByIdElseThrow(textExercise.getId())).thenReturn(textExercise);
+        textExercise = textExerciseRepository.save(textExercise);
+
+        var textParticipation = participationUtilService.createAndSaveParticipationForExercise(textExercise, TEST_PREFIX + "student1");
 
-        textSubmission = new TextSubmission(2L).text("Test - This is what the feedback references - Submission");
+        textSubmission = new TextSubmission().text("Test - This is what the feedback references - Submission");
+        textSubmission.setParticipation(textParticipation);
+        textSubmission = textSubmissionRepository.save(textSubmission);
 
         textBlock = new TextBlock().startIndex(7).endIndex(46).text("This is what the feedback references").submission(textSubmission);
         textBlock.computeId();
-        when(textBlockRepository.findById(textBlock.getId())).thenReturn(Optional.of(textBlock));
+        textBlock = textBlockRepository.save(textBlock);
+
+        var textResult = participationUtilService.addResultToParticipation(textParticipation, textSubmission);
```

Comment:
adding results to participations will be deprecated soon, so this would then need some refactoring. But that's out of scope for this PR